exercises/Mathieu_Tillet/ex01_basics/src/main.cpp

```diff
@@ -0,0 +1,125 @@
+/**
+ * Skeleton main routine
+ */
+
+#include <iostream>
+#include <fstream>
+#include <map>
+#include <vector>
+#include <algorithm>
+#include <tuple>
+
+using namespace std;
+
+int main(int argc, char **argv)
+{
+    map<string, int> word_count;
+
+    if (argc < 2){
+        cerr << "No file provided";
+        return -1;
+    }
+
+    // the input file
+    char *file_name = argv[1];
+
+    string out_name;
+    // Name of the file in which the output will be written
+    if (argc > 2){
+        // catch the second argument if one is provided
+        out_name = string(argv[2]);
+    } else {
+        // else create a new file
+        out_name = "./out.txt";
+    }
+
+    ifstream in;
+    in.open(file_name);
+
+    if (!in){
+        cerr << "Error when reading the file" << endl;
+        return -1;
+    }
+
+    string block;
+    size_t max_length = 4; // size of "Word", which we systematically write
+    size_t block_size;
+    bool should_split;
+
+    // the delimiters of words
+    string delimiters = ".,?'\"!():";
+
+    while (in >> block){
+        // iterate on each space or endofline separated block in the text
```

Comment:
Great to know how to do is manually, but in mantid we have boost available, so I suggest we use `boost::split` to tokenise the strings.